crates/pathfinder/src/state.rs

```diff
@@ -40,167 +42,503 @@ enum UpdateError {
 }
 
 /// Syncs the Starknet state with L1.
-pub async fn sync<T: Transport>(
-    database: &mut Connection,
-    transport: &Web3<T>,
+pub async fn sync<T: Transport + 'static>(
```

Comment:
The issue is that there's too many of the channels and types, but I thought it might be easier crutch to move forward (because it's just a single function/method arg) than designing through a proper unified channel system.